changelog/dmd.deprecate-case-variables.dd

```diff
@@ -0,0 +1,44 @@
+Using run-time variables as switch cases has been deprecated
+
+It's a feature that's not commonly used, and it goes against normal expectations of a switch statement.
+When using run-time variables:
+
+- Case statements can contain duplicates
+- Case statements can be mutated through mutable aliasing of a `const ref`
+- The entire switch statement (including compile time constant cases) generates a series of if-else statements instead of an optimized jump table
```

Comment:
Evidence?

https://d.godbolt.org/z/MeG9drseb suggests otherwise. It's also quite easy to fix so it feels like quite a shoddy argument

Reply:
Also note that jump tables are not magically efficient

Reply:
Technically the truth.  The reality is that smart compilers just see through it.

https://d.godbolt.org/z/M3xsqxrj8

Reply:
i.e. not the truth, no?

A true jump table is not great for your cache so you don't even want one most of the time.